### PR TITLE
Feature/create account

### PR DESCRIPTION
**Pull request by nicolasbarb**

### CODE REVIEW

File: easyselling/Scenario/AccountCreation/AccountCreationViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    
+    private func createAccount(with informations: AccountCreationInformations) {
+        accountCreator.createAccount(informations: informations).sink(receiveCompletion: {
+            if case let .failure(error) = $0 {
+                self.state = .initial
+                self.alert = error
+                self.showAlert = true
+            }
+        }, receiveValue: {
+            self.state = .accountCreated
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AccountCreation/AccountCreationViewModel.swift:49]: Weak self

File: easyselling/Scenario/AccountCreation/AccountCreationViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        case .none: break
+        }
+    }
+    
+    private func createAccount(with informations: AccountCreationInformations) {
+        accountCreator.createAccount(informations: informations).sink(receiveCompletion: {
+            if case let .failure(error) = $0 {
+                self.state = .initial
+                self.alert = error
+                self.showAlert = true
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AccountCreation/AccountCreationViewModel.swift:46]: Weak self

File: easyselling/Scenario/AccountCreation/DefaultAccountCreator.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        self.apiCaller = apiCaller
+    }
+    
+    private var requestGenerator: RequestGenerator
+    private var apiCaller: APICaller
+    
+    func createAccount(informations: AccountCreationInformations) -> VoidResult {
+        let urlRequest = requestGenerator.generateRequest(endpoint: .users, method: .POST, body: informations, headers: [:])
+        guard let urlRequest = urlRequest else {
+            return AnyPublisher(Empty())
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AccountCreation/DefaultAccountCreator.swift:28]: Maybe choose a default error when it failed ??

File: easyselling/Application/Supporting Files/en.lproj/localizable.strings [hunk 4]
```diff
@@ -7,3 +7,7 @@
 */
 
 "helloWorld" = "Hello World ! ";
```
Comment (MaxenceMottard) [inline easyselling/Application/Supporting Files/en.lproj/localizable.strings:9]: Remove

File: easyselling/Application/Supporting Files/fr.lproj/localizable.strings [hunk 5]
```diff
@@ -7,3 +7,7 @@
 */
 
 "helloWorld" = "Bonjour monde !";
```
Comment (MaxenceMottard) [inline easyselling/Application/Supporting Files/fr.lproj/localizable.strings:9]: Remove

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]
Good job 👍🏽

File: easysellingTests/DefaultRequestGenerator_Specs/DefaultRequestGenerator_Specs.swift [hunk 6]
```diff
@@ -40,13 +40,13 @@ class RequestGenerator_Specs: XCTestCase {
...
+        expectedRequest = URLRequest(url: URL(string: "https://easyselling.maxencemottard.com/users")!)
         expectedRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
         expectedRequest.httpMethod = HTTPMethod.POST.rawValue
 
         givenService()
         whenGenerateRequestWithBody(
-            endpoint: "https://easyselling.maxencemottard.com/api/v1/users/profile",
+            endpoint: .users,
             method: .POST,
                                     body: body, headers: [:])
```
Comment (MaxenceMottard) [inline easysellingTests/DefaultRequestGenerator_Specs/DefaultRequestGenerator_Specs.swift:51]: - doesn't require to be on multiple lines
- bad indentation

**Review by PierreGourgouillon — APPROVED** [review]
Good job

**Pull request merged**